test_runner/regress/test_tenant_size.py

```diff
@@ -532,7 +532,24 @@ def get_current_consistent_size(
     assert size_after == prev, "size after restarting pageserver should not have changed"
 
 
-@xfail_on_postgres(PgVersion.V15, reason="Test significantly more flaky on Postgres 15")
+def assert_approx_equal(size_a, size_b):
+    """
+    Tests that evaluate sizes are checking the pageserver space consumption
+    that sits many layers below the user input.  The exact space needed
+    varies slightly depending on postgres behavior.
+
+    Rather than expecting postgres to be determinstic and occasionally
+    failing the test, we permit sizes for the same data to vary by a few pages.
+    """
+
+    # Determined empirically from examples of equality failures: they differ
+    # by page multiples of 8272, and usually by 1-3 pages.
+    threshold = 4 * 8272
```

Comment:
It mentions 1-3 pages, but the multiplicator is 4

Reply:
I'm presuming that if we only ever saw up to 3, then allowing 4 should be safe -- I don't want to set this precisely to 3 and then have to tweak it later.  The number is kind of arbitrary anyway since afaik there is not strictly any bound on what postgres might choose to do

Reply:
Sure, let's mention this in the comment itself

Reply:
Commented updated.